src/category_theory/subobject/comma.lean

```diff
@@ -0,0 +1,218 @@
+  (hq : q ≫ T.map P.arrow = A.hom) : subobject A :=
+subobject.mk (hom_mk P.arrow hq : mk q ⟶ A)
+
+/-- Projecting and then lifting a subobject recovers the original subobject, because there is at
+    most one morphism making the projected subobject into a structured arrow. -/
+lemma lift_project_subobject [has_limits C] [preserves_limits T] {A : structured_arrow S T} :
+  ∀ (P : subobject A) {q} (hq : q ≫ T.map (project_subobject P).arrow = A.hom),
+    lift_subobject (project_subobject P) hq = P := subobject.ind _
+begin
+  introsI P f hf q hq,
+  fapply subobject.mk_eq_mk_of_comm,
+  { fapply iso_mk,
+    { exact subobject.underlying_iso _ },
+    { exact (cancel_mono (T.map f.right)).1 (by { dsimp, simpa [← T.map_comp] using hq }) } },
+  { exact ext _ _ (by { dsimp, simp })}
+end
+
+/-- If `A : S → T.obj B` is a structured arrow for `S : D` and `T : C ⥤ D`, then we can explicitly
+    describe the subobjects of `A` as the subobjects `P` of `B` in `C` for which `A.hom` factors
+    through the image of `P` under `T`. -/
+@[simps]
+def subobject_equiv [has_limits C] [preserves_limits T] (A : structured_arrow S T) :
+  subobject A ≃o { P : subobject A.right // ∃ q, q ≫ T.map P.arrow = A.hom } :=
+{ to_fun := λ P, ⟨project_subobject P, project_subobject_factors P⟩,
+  inv_fun := λ P, lift_subobject P.val P.prop.some_spec,
+  left_inv := λ P, lift_project_subobject _ _,
+  right_inv := λ P, subtype.ext (by simp),
+  map_rel_iff' := subobject.ind₂ _
+  begin
+    introsI P Q f g hf hg,
+    refine ⟨λ h, subobject.mk_le_mk_of_comm _ (ext _ _ _), λ h, _⟩,
+    { refine hom_mk (subobject.of_mk_le_mk _ _ h) ((cancel_mono (T.map g.right)).1 _),
+      simp [← T.map_comp] },
+    { simp only [mono_over.mk'_arrow, subobject.of_mk_le_mk_comp, comma.comp_right, hom_mk_right] },
+    { refine subobject.mk_le_mk_of_comm (subobject.of_mk_le_mk _ _ h).right _,
+      exact congr_arg comma_morphism.right (subobject.of_mk_le_mk_comp h) }
+  end }
+
+/-- If `C` is well-powered and complete and `T` preserves limits, then `structured_arrow S T` is
+    well-powered. -/
+instance well_powered_structured_arrow [well_powered C] [has_limits C] [preserves_limits T] :
+  well_powered (structured_arrow S T) :=
+{ subobject_small := λ X, small_map (subobject_equiv X).to_equiv }
+
+end structured_arrow
+
+namespace costructured_arrow
+variables {S : C ⥤ D} {T : D}
+
+/-- Every quotient of a costructured arrow can be projected to a quotient of the underlying
+    object. -/
+def project_quotient [has_colimits C] [preserves_colimits S] {A : costructured_arrow S T} :
+  subobject (op A) → subobject (op A.left) :=
+begin
+  refine subobject.lift (λ P f hf, by exactI subobject.mk f.unop.left.op) _,
+  introsI P Q f g hf hg i hi,
+  refine subobject.mk_eq_mk_of_comm _ _ ((proj S T).map_iso i.unop).op (quiver.hom.unop_inj _),
+  have := congr_arg quiver.hom.unop hi,
+  simpa using congr_arg comma_morphism.left this,
```

Comment:
I guess you can combine these lines?

Reply:
Unfortunately, this leads to a type mismatch error. I don't know why.